libbpf-tools/gethostlatency.bpf.c

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2021 Hengqi Chen
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "gethostlatency.h"
+
+#define MAX_ENTRIES 10240
+
+const volatile pid_t targ_tgid = 0;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, __u32);
+	__type(value, struct val_t);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(value_size, sizeof(__u32));
+} events SEC(".maps");
+
+static __always_inline
+int probe_entry(struct pt_regs *ctx) {
+	if (!PT_REGS_PARM1(ctx))
+		return 0;
+
+	struct val_t val = {};
+	__u32 pid = bpf_get_current_pid_tgid() >> 32;
```

Comment:
You changed this from the last version, right? I think tid is also OK. Each thread can have its own comm so it might be confusing to print out tgid instead of tid.

Reply:
Yes, I changed it.

There are more than 70 tools in this repo use `bpf_get_current_pid_tgid` and they expose option named `PID`.

43 of them make explicit conversion using `bpf_get_current_pid_tgid >> 32`, the rest simply using something like `__u32 pid = bpf_get_current_pid_tgid()`, I suspect some of them misuse the `bpf_get_current_pid_tgid` helper.